huaweicloud/services/nat/resource_huaweicloud_nat_private_dnat_rule.go

```diff
@@ -0,0 +1,232 @@
+package nat
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk/openstack/nat/v3/dnats"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourcePrivateDnatRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePrivateDnatRuleCreate,
+		ReadContext:   resourcePrivateDnatRuleRead,
+		UpdateContext: resourcePrivateDnatRuleUpdate,
+		DeleteContext: resourcePrivateDnatRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the DNAT rule is located.",
+			},
+			"gateway_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The private NAT gateway ID to which the DNAT rule belongs.",
+			},
+			"transit_ip_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the transit IP for private NAT.",
+			},
+			"transit_service_port": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The port of the transit IP.",
+			},
+			"protocol": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The protocol type.",
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
should we remove the validate func

Reply:
all validate functions have been deleted.